### DISCUSSION TITLE
Is there any way to store float values for the calculation purpose on the chain and define in the contract ?

### DISCUSSION THREAD

**Discussion by Nishu0** [aptos-labs/aptos-developer-discussions#201] [Questions]
### Discord user ID

itznisargthakkar

### Describe your question in detail.

I have written a contract to provide loans to borrowers based on the RWA staking worth. Now, on the pool creation, I am defining the interest rate at which the borrower needs to repay the loan amount. At the pool creation I am doing the expected amount calculation which is the float value and the interest rate is also the float value. I am providing the option to repay the loan on an EMI basis and for storing the expected EMI I also need the float value. So there any way to store the float value for the calculation purpose? in the Move as Well on the Chain 

### What have you tried or looked at?  Or how can we reproduce the error?

Whenever I do a calculation it will convert the float value to an integer automatically I know the type is an Unsigned Integer itself but need a type to store float values. Can you provide any example of how to store and do calculation on the float values?

### Which operating system are you using?

Windows

### Which SDK or tool are you using? (if any)

TypeScript SDK

**Comment by gregnazario** [comment]
You can take a look at `FixedPoint` for your calculations https://aptos.dev/reference/move/?branch=mainnet&page=aptos-stdlib/doc/fixed_point64.md

That'll do math with a known fraction, rather than floating point.

Here's an example https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-stdlib/tests/fixedpoint64_tests.move#L77-L82
